query/src/sql/executor/physical_plan_builder.rs

```diff
@@ -78,21 +83,24 @@ impl PhysicalPlanBuilder {
                             .collect::<Result<Vec<_>>>()
                     })
                     .transpose()?;
-
-                let mut source = metadata.table(scan.table_index).source.clone();
-                source.push_downs = push_down_filters.map(|filters| Extras {
-                    filters,
+                
+                let projection = scan.columns.iter().map(|i| *i).sorted().collect::<Vec<_>>();
```

Comment:
The `scan.columns` is not exactly the column indices that can be pushed down, but the column indices within `Metadata`.

I think we need to remap the column index to offset of the column inside the table with `name_mapping` and schema of the `Table`.